src/test/run-make-fulldeps/rustdoc-io-error/Makefile

```diff
@@ -0,0 +1,22 @@
+-include ../tools.mk
+
+# This test verifies that rustdoc doesn't ICE when it encounters an IO error
+# while generating files. Ideally this would be a rustdoc-ui test, so we could
+# verify the error message as well.
+
+OUTPUT_DIR := "$(TMPDIR)/rustdoc-io-error"
+
+# Ignore Windows: the test uses `chmod`.
+ifndef IS_WINDOWS
+
+# This test operates by creating a temporary directory and modifying its
+# permissions so that it is not writable. We have to take special care to set
+# the permissions back to normal so that it's able to be deleted later.
+all:
+	mkdir -p $(OUTPUT_DIR)
+	chmod u-w $(OUTPUT_DIR)
+	-$(shell $(RUSTDOC) -o $(OUTPUT_DIR) foo.rs)
```

Comment:
Does this save the exit status into `$(.SHELLSTATUS)` for later? I'm not familiar with this syntax.

Reply:
Yes. https://www.gnu.org/software/make/manual/html_node/Shell-Function.html#Shell-Function